use-record.js

```diff
@@ -1,28 +1,41 @@
-import React from "react"
+import React from 'react'
 import { Deepstream } from './provider'
+import { Observable } from 'rxjs'
+
+export const useRecord = (ids, path, state, debounce) => {
+  let initialValue = null
+  if (Array.isArray(ids)) {
+    initialValue = []
+  } else if (!path) {
+    initialValue = {}
+  }
 
-export const useRecord = (id, path) => {
   const { ds, auth } = React.useContext(Deepstream)
-  const [data, setData] = React.useState(null)
+  const [values, setValues] = React.useState(initialValue)
+
+  ids = auth && ds && ids && ids.length > 0 ? ids : null
 
   React.useEffect(() => {
-    if (!ds || !auth) {
-      return
-    }
+    if (!ids) {
+      setValues(initialValue)
+    } else {
+      const observe = id => ds.record
+        .observe(id, state)
+        .publish(x$ => path ? x$.pluck(...path.split('.')).distinctUntilChanged() : x$)
+
+      const observable = Array.isArray(ids)
+        ? Observable.combineLatest(ids.map(observe))
+        : observe(ids)
+
+      const subscription = observable
+        .publish(x$ => typeof debounce === 'number' ? x$.debounceTime(debounce) : x$)
+        .subscribe(setValues)
 
-    const subscription = ds.record.observe(id)
-      .subscribe(data => {
-        if (path) {
-          setData(data[path])
-        } else {
-          setData(data)
-        }
-      })
-
-    return () => {
-      subscription.unsubscribe()
+      return () => {
+        subscription.unsubscribe()
+      }
     }
-  }, [ds, auth, id, path])
+  }, [ds, path, state, debounce].concat(ids))
```

Comment:
I can't find anywhere in the docs whether this array is allowed to have a dynamic size or not...